homeassistant/components/surepetcare/sensor.py

```diff
@@ -89,8 +109,125 @@ def _async_update(self) -> None:
             }
         else:
             self._attr_extra_state_attributes = None
+
```

Comment:
Seems to be a lot of duplicated code here.
Possible to make a more generic SureSensorEntity ?

Reply:
are you kidding me? :D there was a SureSensorEntity until I removed it to reduce the size of this pr lol :D

in general, I absolutely agree and its already implemented 👍  but I would tend to doing this in the next pr

Reply:
Duplicating code does sound like a good way to reduce the size of the pr.
Then I think it is better to remove all the unrelated changes to _LOGGER.debug(